synapse/models/infotech.py

```diff
@@ -25,559 +24,770 @@ class SemverType(DataType):
-                    ('path:ext', {'ptype': 'str:lwr', 'doc': 'The file extension of the file name (parsed from :path).', 'ro': 1}),
-                    ('path:base', {'ptype': 'file:base', 'doc': 'The final component of the file path (parsed from :path).', 'ro': 1}),
-                    ('file', {'ptype': 'file:bytes', 'doc': 'The file that was read.'}),
+                ('it:exec:url', {}, (
+                    ('proc', ('it:exec:proc', {}), {
+                        'doc': 'The main process executing code that requested the URL.',
+                    }),
+                    ('host', ('it:host', {}), {
+                        'doc': 'The host running the process that requested the URL. Typically the same host referenced in :proc, if present.',
+                    }),
+                    ('exe', ('file:bytes', {}), {
+                        'doc': 'The specific file containing code that requested the URL. May or may not be the same :exe specified in :proc, if present.',
+                    }),
+                    ('time', ('time', {}), {
+                        'doc': 'The time the URL was requested.',
+                    }),
+                    ('url', ('inet:url', {}), {
+                        'doc': 'The URL that was requested.',
+                    }),
+                    # FIXME Needs inet:addr
+                    # ('client', ('inet:addr', {}), {
+                    #     'doc': 'The address of hte client during the URL retrieval.'
+                    # })
+                    # FIXME - Break out inet:addr subs!
                 )),
-
-                ('it:exec:file:write', {'ptype': 'guid'}, (
-                    ('proc', {'ptype': 'it:exec:proc', 'doc': 'The main process executing code that wrote to / modified the existing file.'}),
-                    ('host', {'ptype': 'it:host',
-                         'doc': 'The host running the process that wrote to the file. Typically the same host referenced in :proc, if present.'}),
-                    ('exe', {'ptype': 'file:bytes',
-                         'doc': 'The specific file containing code that wrote to the file. May or may not be the same :exe referenced in :proc, if present.'}),
-                    ('time', {'ptype': 'time', 'doc': 'The time the file was written to / modified.'}),
-                    ('path', {'ptype': 'file:path', 'doc': 'The path where the file was modified.'}),
-                    ('path:dir', {'ptype': 'file:path', 'doc': 'The parent directory of the file path (parsed from :path).', 'ro': 1}),
-                    ('path:ext', {'ptype': 'str:lwr', 'doc': 'The file extension of the file name (parsed from :path).', 'ro': 1}),
-                    ('path:base', {'ptype': 'file:base', 'doc': 'The final component of the file path (parsed from :path).', 'ro': 1}),
-                    ('file', {'ptype': 'file:bytes', 'doc': 'The file that was modified.'}),
+                ('it:exec:bind', {}, (
```

Comment:
:-D nice that we can get rid of bind:tcp, etc

Reply:
:)